stl/inc/algorithm

```diff
@@ -5634,9 +5634,84 @@ _NODISCARD bool is_sorted(_ExPo&& _Exec, _FwdIt _First, _FwdIt _Last) noexcept /
     return _STD is_sorted(_STD forward<_ExPo>(_Exec), _First, _Last, less{});
 }
 
-#endif // _HAS_CXX17
+#ifdef __cpp_lib_concepts
+namespace ranges {
+    // FUNCTION TEMPLATE _Is_sorted_until_unchecked
+    template <class _It, class _Se, class _Pr, class _Pj>
+    _NODISCARD constexpr _It _Is_sorted_until_unchecked(_It _First, const _Se _Last, _Pr _Pred, _Pj _Proj) {
+        _STL_INTERNAL_STATIC_ASSERT(forward_iterator<_It>);
+        _STL_INTERNAL_STATIC_ASSERT(sentinel_for<_Se, _It>);
+        _STL_INTERNAL_STATIC_ASSERT(indirect_strict_weak_order<_Pr, projected<_It, _Pj>>);
+
+        if (_First == _Last) {
+            return _First;
+        }
+
+        for (auto _Prev = _First; ++_First != _Last; ++_Prev) {

```

Comment:
Observation: This increments `_Prev` instead of assigning `_Prev = _First`. Could the increment be more expensive than an assignment (e.g. linked lists)? Maybe not, due to how caching works. I also observe that the classic algorithm performs independent increments. No change or issue requested.

Reply:
I pondered this a bit as well while hacking up the code after copying from `std`. I can think of plenty iterators for which copy is more expensive than increment, but few for which increment is more expensive than copy. Copy almost always touches Θ(sizeof(i)) state, but increment often updates only part of an iterator's state. (Again assuming that any required pointer chasing is cached.)

Given that my intuition agrees with the pre-existing code, I'm happy.